### ISSUE TITLE
[DOC] Ambiguity regarding exog data documentation

### ISSUE THREAD

**Issue by IlyasMoutawwakil** [alan-turing-institute/sktime#1462]
> #### Describe the issue linked to the documentation

The ForecastGorizon interface allows seperated steps forecasting (like `fh=[-10, -5, 3]`) but the forecaster (like pmdarima) is performing in this case a forecast from -10 to 3. The documentation says:

```
X: pd.DataFrame, optional (default=None)
Exogeneous data
```
But it doesn't explain that X should have all exogenous values from relative step -10 to 3 (because that's what pmdarima is expecting).

**Comment by fkiraly** [comment]
Yes, I agree it's an interesting issue - my original answer from #1442 (where this discussion cropped up off-topic):

@IlyasMoutawwakil, that's an interesting point - there's actually two related things here:

* telling the user about the requirement to pass all indices between start/end
* representing the requirement.
* whether we can get around the behaviour

For the first thing, we are using tags, which you can list by `registry.all_tags`.
We cannot use the `_fit` etc docstring directly, since the user facing docstring is the `fit` in the base class - but we probably should state assumptions in the docstrings, related to the tags. I've opened an issue here #1451.

Regarding tags:
For forecasters, we have `"X-y-must-have-same-index"` which covers the case where `X` indices need to contain those in `y` (being forecast or observed).
This does not cover the above case, as you state. So I wonder how we should indicate this as a tag, or whether we should change the meaning of the existing tag. Or, otherwise, how we tell the user.

Any good ideas, @IlyasMoutawwakil ?

**Comment by fkiraly** [comment]
PS: I have no better idea than another tag such as `"X-must-have-all-periodic-indices-until-most-recent-y-index"` or similar (perhaps less verbose), document it, and include it in the forecaster docstrings too (since it implies an assumption on the inputs for forecasters that have this)
